test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -721,50 +745,47 @@
                     certname]
                    {:as-arrays? true}))))))))
 
-  (deftest facts-blacklist
+  (deftest regex-facts-blacklist
     (dotestseq [version fact-versions
-                :let [command (update facts :values #(assoc % "blacklisted-fact" "val"))]]
-      (testing "should ignore the blacklisted fact"
+                :let [command (update facts :values
+                                      #(assoc %
+                                              "blacklisted-fact" "val"
+                                              "prefix" "val"
+                                              "notprefix" "val"
+                                              "facts-suff" "val"
+                                              "glob" "val"
+                                              "puppet" "val"))]]
+      (testing "should ignore the blacklisted facts using regex"
         (with-message-handler {:keys [handle-message dlo delay-pool q]}
           (handle-message (queue/store-command q (facts->command-req (version-kwd->num version) command)))
-          (is (= (query-to-vec
-                  "SELECT fp.path as name,
-                          COALESCE(fv.value_string,
-                                   cast(fv.value_integer as text),
-                                   cast(fv.value_boolean as text),
-                                   cast(fv.value_float as text),
-                                   '') as value,
-                          fs.certname
-                   FROM factsets fs
-                     INNER JOIN facts as f on fs.id = f.factset_id
-                     INNER JOIN fact_values as fv on f.fact_value_id = fv.id
-                     INNER JOIN fact_paths as fp on f.fact_path_id = fp.id
-                   WHERE fp.depth = 0
-                   ORDER BY name ASC")
+          (is (= (approx-facts-query)
                  [{:certname certname :name "a" :value "1"}
                   {:certname certname :name "b" :value "2"}
-                  {:certname certname :name "c" :value "3"}]))))))
+                  {:certname certname :name "c" :value "3"}
+                  {:certname certname :name "notprefix" :value "val"}]))))))
+
+    (deftest literal-facts-blacklist
```

Comment:
Wondering why the diff thinks this function is completely new -- I suspect it might be ordering.  If you feel like it, might try putting this test before the regex test and see if that shrinks the diff (and makes what really happened clearer).

Reply:
Nevermind - forgot about indentation changes, if nothing else.